### ISSUE TITLE
windows alternate tasks

### ISSUE THREAD

**Issue by jdx** [jdx/mise#2595]
because shell commands work quite differently in windows, we could allow configuration for tasks specific to OS, such as:

```toml
[tasks.echo]
run = 'echo $FOO'
run_windows = 'echo %FOO%'
```

or maybe something like this:

```toml
[tasks.echo]
run = 'echo $FOO'
[tasks-windows.echo]
run = 'echo %foo%'
```

I think the former is what I prefer, though I can think of a few different ways to present this like maybe a `tasks.os.windows` key, though that would prevent anyone from having a task labeled "os" so it would have drawbacks.

**Comment by rodvieirasilva** [comment]
I am using Mise on Windows, and I encountered some difficulties with shell command compatibility. To address the differences in shell commands across operating systems, I propose adding an os.windows/os.unix section in the TOML configuration that can override any configuration. This approach allows users to specify OS-specific settings not only for tasks but for other sections as well, such as [settings], [tools], or any other sections that may benefit from customization.

The configuration would work as follows: when an OS-specific section (e.g., [os.windows]) is provided, it will inherit properties from the main configuration, but any properties specified in the OS-specific section will override the main configuration values. This enables users to customize tasks and tools with a high degree of flexibility while keeping platform-specific variations organized and easy to manage.

Example of Proposed Structure:
Base Configuration:
```toml
[tasks.clean]
description = "Clean the build directory before building"
run = '''
    echo "Cleaning..." > clean.log
'''
```
OS-Specific Override:

```toml
[os.windows]
[tasks.clean]
description = "Windows-specific clean task"
shell = "PowerShell" # specify shell for Windows

[os.unix]
[tasks.clean]
shell = "bash" # specify shell for Unix
```
Benefits:
Consistency and Clarity: Each task can inherit tags and configurations from the base [tasks] section, while also applying OS-specific overrides as needed.
Flexibility: Beyond tasks, any other configuration sections (e.g., [settings], [tools]) can be similarly customized with platform-specific values without impacting the main configuration.
Ease of Use: This structure keeps OS-specific commands and settings consolidated, making it straightforward to understand and modify configurations for different environments.
This approach balances flexibility and clarity, allowing projects to handle platform-specific configurations more seamlessly while minimizing redundancy.

If I receive an OK with this proposal, I can implement the issue.

Thanks.

**Comment by jdx** [comment]
you could do this today with `MISE_ENV=windows` and have a `mise.windows.toml` file

**Comment by hverlin** [comment]
I think it's not working correctly currently?
<img width="731" alt="image" src="https://github.com/user-attachments/assets/333f4504-e7f2-4290-8011-e6ef8efaf374">

How is the merge done also? Do we expect the task to be fully redefined or merged?
How does one unset a value also?

**Comment by rodvieirasilva** [comment]
I’ve opened a https://github.com/jdx/mise/pull/2967 that enables the execution of scripts on Windows. We can discuss it, and I’ll make any necessary changes.

**Comment by jdx** [comment]
> I think it's not working correctly currently? <img alt="image" width="731" src="https://private-user-images.githubusercontent.com/9151470/384218816-333f4504-e7f2-4290-8011-e6ef8efaf374.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzExNjU4NzUsIm5iZiI6MTczMTE2NTU3NSwicGF0aCI6Ii85MTUxNDcwLzM4NDIxODgxNi0zMzNmNDUwNC1lN2YyLTQyOTAtODAxMS1lNmVmOGVmYWYzNzQucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MTEwOSUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDExMDlUMTUxOTM1WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9ZDAwNTY1ZGNjY2UzYTljMTM1MDFlZjZlMGYyMjFiZGYwZThhOWQ5ZmIzNjZiNzU0NzBiMjBiYzFhMGZkMDUwMCZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ._zaa0GTz37ltj-SabM2wOLgZALi9BCbLkMju8XQAurE">
> 
> How is the merge done also? Do we expect the task to be fully redefined or merged? How does one unset a value also?

this bug should be fixed in #2973
